### ISSUE TITLE
Cannot fetch They Said So quotes due to CORS issue

### ISSUE THREAD

**Issue by Ray-Eldath** [joelshepherd/tabliss#601]
Don't know why this happened. But seems They Said So keeps returning CORS error and a 401 Unauthorized.

![image](https://user-images.githubusercontent.com/7938763/233410438-a3019d74-fdde-4e3e-aab0-7bee8eeced7d.png)
![image](https://user-images.githubusercontent.com/7938763/233405985-c2188dc7-e017-47d0-81e3-d7a5b25c09b6.png)

**Comment by Mv0sKff** [comment]
Same problem on Firefox.
Looks like they changed the api

**Comment by jeremylaier2** [comment]
Not a cors issue. Tabliss API key is expired.

**Comment by Mv0sKff** [comment]
Well, I made a hacky fork that shows a quote from a couple of hard coded ones that I copied from https://github.com/dwyl/quotes **for myself**.
There is a release for Chrome and Firefox (unsigned, as of right now)
https://github.com/Mv0sKff/tabliss_offline_quotes

Would be great if this issue would be addressed

**Comment by marcodussin** [comment]
Latest merge is 4 months ago. Time to fork the project?

**Comment by lupus0802** [comment]
I'm currently trying to add the functionality of supplying an API key in .env and then including that when building the extension in my fork. My initial plan was then to post a explanation on how to build and migrate the extension but there's still one issue.

The code in itself is already done, however, I'm still running into the CORS issue. When looking at the Preflight response headers, it seems like the API is not sending an Access Control Header when sending a 401 response even though they state it on their website, thus the browser not even sending the actual request with the auth header. Please correct me if I'm wrong in this case.

I've reached out to the TheySaidSo API via their Contact form, I guess we'll have to wait for them to change something.

**Comment by maikol-solis** [comment]
> Well, I made a hacky fork that shows a quote from a couple of hard coded ones that I copied from https://github.com/dwyl/quotes **for myself**. There is a release for Chrome and Firefox https://github.com/Mv0sKff/tabliss_offline_quotes

Hi, I tried to install your version, but I got this error. 

![image](https://github.com/joelshepherd/tabliss/assets/4078489/baa6275d-108a-4004-9054-b594582d56bc)

Thanks.

**Comment by Mv0sKff** [comment]
> > Well, I made a hacky fork that shows a quote from a couple of hard coded ones that I copied from https://github.com/dwyl/quotes **for myself**. There is a release for Chrome and Firefox https://github.com/Mv0sKff/tabliss_offline_quotes
> 
> Hi, I tried to install your version, but I got this error.
> 
> ![image](https://user-images.githubusercontent.com/4078489/240594304-baa6275d-108a-4004-9054-b594582d56bc.png)
> 
> Thanks.

Try installing it from here https://addons.mozilla.org/en-US/firefox/addon/tabliss-offline-quotes/

**Comment by maikol-solis** [comment]
@Mv0sKff It works flawlessly. Thanks!

However, I'm confused because all the categories in the settings sidebar aren't really used, or am I wrong? 

![Image](https://github.com/joelshepherd/tabliss/assets/4078489/25838aae-0524-44d6-850b-40287021b3f4)

**Comment by Mv0sKff** [comment]
No, you are right.
Choosing a category does nothing.

These are all the quotes that show up at random: [src/plugins/widgets/quote/quotes.ts](https://github.com/Mv0sKff/tabliss_offline_quotes/blob/9dd7529c19c2acc857cd9425be81d7f66af27caf/src/plugins/widgets/quote/quotes.ts)

**Comment by maikol-solis** [comment]
Ok perfect. The add-on works perfectly fine. I just wanted to be sure about this.

Thanks to revive this feature.

**Comment by the-wright-jamie** [comment]
> Latest merge is 4 months ago. Time to fork the project?

Yeah, last update to stores was 2022, I forked and am going through updating packages and rebranding for a re-release. Anyone have any ideas or feature suggestions or pull requests they'll definitely be seen by an active dev

https://github.com/the-wright-jamie/tab-nine

~~Working on some changes so it's not pretty right now and no release but I'm hoping to clean up some of the bit rot~~

Release available: https://github.com/the-wright-jamie/tab-nine#download
